res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2149 @@
+        ledNumber,
+        active ? 0x7F : 0x00
+    );
+};
+
+PioneerDDJSX.updateParameterStatusLeds = function(group, statusRoll, statusSlicer, statusSampler) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.parameterLeftRollMode, statusRoll & (1 << 1));
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.parameterRightRollMode, statusRoll & 1);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.parameterLeftSlicerMode, statusSlicer & (1 << 1));
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.parameterRightSlicerMode, statusSlicer & 1);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.parameterLeftSamplerMode, statusSampler & (1 << 1));
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.parameterRightSamplerMode, statusSampler & 1);
+};
+
+
+///////////////////////////////////////////////////////////////
+//                             LEDS                          //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSX.fxAssignLeds = function(value, group, control) {
+    var unit = PioneerDDJSX.fxUnitGroups[group],
+        ledNumber = PioneerDDJSX.fxAssign[control];
+
+    PioneerDDJSX.fxAssignLedControl(unit, ledNumber, PioneerDDJSX.shiftPressed, value);
+};
+
+PioneerDDJSX.fxLeds = function(value, group, control) {
+    var unit = 0,
+        searchStr,
+        ledNumber = 0;
+
+	for (var index in PioneerDDJSX.fxUnitGroups) {
+		if (PioneerDDJSX.fxUnitGroups.hasOwnProperty(index)) {
+			searchStr = index.replace(']', '_');
+			if (group.indexOf(searchStr) !== -1) {
+				unit = PioneerDDJSX.fxUnitGroups[index];
+				ledNumber = PioneerDDJSX.fxEffectGroups[group];
+			} else if (group.indexOf(index) !== -1) {
+				unit = PioneerDDJSX.fxUnitGroups[index];
+				ledNumber = 0x03;
+			}
+		}
+	}
+
+    PioneerDDJSX.fxLedControl(unit, ledNumber, PioneerDDJSX.shiftPressed, value);
+    PioneerDDJSX.fxLedControl(unit, ledNumber, true, value);
+};
+
+PioneerDDJSX.headphoneCueLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.headphoneCue, value);
+};
+
+PioneerDDJSX.shiftHeadphoneCueLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftHeadphoneCue, value);
+};
+
+PioneerDDJSX.shiftMasterCueLed = function(value, group, control) {
+    PioneerDDJSX.generalLedControl(PioneerDDJSX.nonPadLeds.shiftMasterCue, value);
+};
+
+PioneerDDJSX.keyLockLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.keyLock, value);
+};
+
+PioneerDDJSX.playLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.play, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftPlay, value);
+};
+
+PioneerDDJSX.playFromCuePoint = false;
+
+PioneerDDJSX.playIllumination = function(value, group, control) {
```

Comment:
After checking the different cue modes set in the Mixxx's preferences, I can say that there's no problem concerning the illumination control/ jog wheel LEDs. The illumination control is independent of the cue mode. A better way would be to be able to control the jog wheel LEDs itself, but as already said that would take a lot of rework (without using "Serato Mode").